File: Android/client/lib/api/customers.dart
import 'api_client.dart';

class CustomersApi {
  static final _api = BizOsApiClient.instance;

  static Future<List<Map<String, dynamic>>> list({
    Map<String, dynamic>? query,
  }) async => _api.list(await _api.get('/client/customers', query: query));
  static Future<Response<dynamic>> create(Map<String, dynamic> data) =>
      _api.post('/client/customers', data: data);
  static Future<Response<dynamic>> get(String id) =>
      _api.get('/client/customers/$id');
  static Future<Response<dynamic>> update(
    String id,
    Map<String, dynamic> data,
  ) => _api.patch('/client/customers/$id', data: data);
  static Future<Response<dynamic>> remove(String id) =>
      _api.delete('/client/customers/$id');
}

File: Desktop/src/api/customers.ts
import { api } from './axios';
import type { Customer, CreateCustomerInput } from '@/types/customer';
import type { ApiPaginated } from '@/types/api';

export const customerApi = {
  list: (params: { page?: number; limit?: number; search?: string } = {}) =>
    api
      .get<ApiPaginated<Customer>>('/client/customers', { params })
      .then((r) => r.data),

  get: (id: string) =>
    api.get<{ data: Customer }>(`/client/customers/${id}`).then((r) => r.data.data),

  create: (payload: CreateCustomerInput) =>
    api.post<{ data: Customer }>('/client/customers', payload).then((r) => r.data.data),

  update: (id: string, payload: Partial<CreateCustomerInput>) =>
    api
      .patch<{ data: Customer }>(`/client/customers/${id}`, payload)
      .then((r) => r.data.data),

  remove: (id: string) => api.delete(`/client/customers/${id}`).then((r) => r.data),
};